target/scripts/helper-functions.sh

```diff
@@ -288,27 +288,64 @@ export -f _monitored_files_checksums
 
 # ? --------------------------------------------- General
 
+# Outputs the DNS label count (delimited by `.`) for the given input string.
+# Useful for determining an FQDN like `mail.example.com` (3), vs `example.com` (2).
+function _get_label_count
+{
+  awk -F '.' '{ print NF }' <<< "${1}"
+}
+
+# Sets HOSTNAME and DOMAINNAME globals used throughout the scripts,
+# and any subprocesses called that intereact with it.
 function _obtain_hostname_and_domainname
 {
-  if [[ -n "${OVERRIDE_HOSTNAME}" ]]
+  # Normally this value would match the output of `hostname` which mirrors `/proc/sys/kernel/hostname`,
+  # However for legacy reasons, the system ENV `HOSTNAME` was replaced here with `hostname -f` instead.
+  #
+  # TODO: Consider changing to `DMS_FQDN`; a more accurate name, and removing the `export`, assuming no
+  # subprocess like postconf would be called that would need access to the same value via `$HOSTNAME` ENV.
+  #
+  # TODO: `OVERRIDE_HOSTNAME` was introduced for non-Docker runtimes that could not configure an explicit hostname.
+  # k8s was the particular runtime in 2017. This does not update `/etc/hosts` or other locations, thus risking
+  # inconsistency with expected behaviour. Investigate if it's safe to remove support. (--net=host also uses this as a workaround)
+  export HOSTNAME="${OVERRIDE_HOSTNAME:-$(hostname -f)}"
```

Comment:
I would not like to see `OVERRIDE_HOSTNAME` being removed as is without introducing another variable, maybe for K8s, that does an equal job. I'd be very hesitant to remove this. Maybe adjust the comment, or remove it at all?

Reply:
>  Maybe adjust the comment, or remove it at all?

The comment stays. It makes it rather clear what the concern is and what needs to be done for it to be considered for removal:

- Introduced for non-Docker runtimes that could not configure an explicit hostname (_k8s was the particular runtime in 2017_). `--net=host` also uses this as a workaround.
- This does not update `/etc/hosts` or other locations, thus risking inconsistency with expected behaviour.
- Investigate if it's safe to remove support.

I'm likely one of the only ones who will investigate it, and that will be a while off as it requires big adjustments to the test suite that I'm slowly chipping away at.

If you can confirm that in almost 2022, this remains a requirement for k8s that is helpful :)

---

> I would not like to see `OVERRIDE_HOSTNAME` being removed as is without introducing another variable

I know of one case with `--network=host` where it is currently required, but I don't consider the workaround satisfactory due to the lack of `/etc/hosts` parity.

What is likely to happen is the tests should show if the `/etc/hosts` addition matters at all, other than `hostname -f` or similar commands or if we can shift away from `hostname`/`domainname` config and have all scenarios unify on a common ENV instead.

If that makes sense to do, it can't be done confidently AFAIK without the test matrix run I want to be able to perform :sweat_smile:

The only way `OVERRIDE_HOSTNAME` will get removed or replaced is via PR anyway, I see no harm in the comment, only additional clarification to future maintainers :)

Reply:
Alright, we keep the comment, this is fine with me :)